tests/pipeline/test_pipeline_delegator.py

```diff
@@ -46,13 +47,17 @@ def callback_func(args: List[Tuple[str, QueryResult]]) -> None:
     delegator = PipelineDelegator(
         query_pipeline_builders={"events": events_pipeline, "errors": errors_pipeline},
         selector_func=lambda query: ("events", ["errors"]),
-        callback_func=callback_func,
+        callback_func=mock_callback,
     )
 
     with cv:
         delegator.build_pipeline(
             Request("", query, HTTPRequestSettings(), {}, ""), mock_query_runner
         ).execute()
-        cv.wait()
+        cv.wait(timeout=5)
```

Comment:
Please verify that the return is True. Which means this did not end for timeout. So you can just immediately raise if the return is False, which makes the test fail.

Reply:
Any issue just letting this pass even when this times out? After all this is known to be flaky and could end up being rather annoying. As long as the assertion on line 61 is ok, I think the test should be allowed to pass.

Reply:
I believe the only thing that change would be testing is the behaviour of wait() and notify() which we know to be flaky

Reply:
the difference is that it would fail right away. But merge as it is is still better than nothing. 
We need to investigate anyway.